### DISCUSSION TITLE
Would it make sense to add LDAP support?

### DISCUSSION THREAD

**Discussion by vromero** [gotson/komga#1417] [Ideas]
Hello, 

I have just recently dabbled into this project and configured it with OIDC with authentik.

I read in another issue that the removal of local users is a non-starter due to some design choice, which of course is fair enough.

I would like, however, to propose a use-case and a potential solution. 

**Use case:** Someone has emby (which is similar to komga in which there are client apps that do not support OIDC) alongside komga, then also some identity service like authentik or authelia, the desire is to have only one user that can operate in both systems with only one password.

**Solution:** A perhaps valid solution for this would be to support OIDC and LDAP at the same time. OIDC would take care of the website login while if some third party app tries to use the basic login it will still be able to do so because LDAP can connect to the same source of identity. The user would not feel they have multiple logins but one. Emby, as an example, does support LDAP through the LDAP plugin (I currently have it configured like that).

```mermaid
graph LR;
    
    Emby --> AuthentikLDAPGateway;
    Komga --> AuthentikLDAPGateway;
    Komga --> AuthentikOIDCGateway;

 AuthentikLDAPGateway --> Authentik;
 AuthentikOIDCGateway --> Authentik;
```

The beauty of this solution is that requires no development, all it would _probably_ (honestly havent tried) take is to add some dependencies, because the user can already override the spring configuration through `appplication.yml`.

```
implementation("org.springframework.boot:spring-boot-starter-security")
implementation("org.springframework.ldap:spring-ldap-core")
implementation("org.springframework.security:spring-security-ldap")
```

**Comment by gotson** [comment]
There's no plans to add LDAP. You can use OAuth2 which is already there.

**Comment by vromero** [comment]
I do use OAuth2 in my current test installation. I hope though my point came across correctly as OAuth2/OIDC do not fulfill the use-case I brought up.

This said, no plans is fair enough. Thanks.
